.ci/ci-build.sh

```diff
@@ -26,6 +26,7 @@ list_commits  || failure 'Could not detect added commits'
 list_packages || failure 'Could not detect changed files'
 message 'Processing changes' "${commits[@]}"
 test -z "${packages}" && success 'No changes in package recipes'
+define_build_order || failure 'Could not determine build order'
```

Comment:
So far so good. What changed about the hang?

Reply:
The fact that I found out _why_ it "hung": I failed to realize that `actions/checkout@v2` performs a shallow clone by default, which made `git log upstream/main..` look as if _every single package_ was updated, and hence `define_build_order` (which is quadratic in the number of packages, if I remember correctly) worked accordingly hard... 😁

Reply:
That explains the unexpectedly long time to run.